src/mysky.ts

```diff
@@ -304,6 +317,44 @@ export class MySky {
   // Internal Methods
   // ================
 
+  /**
+   * Set up a listener for the storage event. If the seed is set in the UI, it
+   * should trigger a load of the permissions provider.
+   */
+  setUpStorageEventListener(): void {
+    window.addEventListener("storage", ({ key, newValue }: StorageEvent) => {
+      if (key !== SEED_STORAGE_KEY) {
+        return;
+      }
+
+      if (this.permissionsProvider) {
+        // Unload the old permissions provider. No need to await on this.
+        void this.permissionsProvider.then((provider) => provider.close());
+        this.permissionsProvider = null;
+      }
+
+      if (!newValue) {
+        // Seed was removed.
+        return;
+      }
+
+      // Parse the seed.
+      const seed = new Uint8Array(JSON.parse(newValue));
+
+      // Launch the new permissions provider.
+      this.permissionsProvider = launchPermissionsProvider(seed);
+
+      // If JWT is found, then set it on the client.
+      const jwt = localStorage.getItem(JWT_STORAGE_KEY);
+      if (jwt) {
+        // Clear the stored JWT.
+        localStorage.removeItem(JWT_STORAGE_KEY);
```

Comment:
It's not entirely clear to me why we're removing the JWT from `localStorage`.

Reply:
`localStorage` is only used to marshal the JWT from MySky UI over to the invisible MySky iframe. For the seed, we keep it in storage so that users are automatically logged-in -- but for the JWT, it should rather be stored somewhere on MySky, as the local storage can get cleared, users can move across browsers etc.